packages/canon/src/components/button/button.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { button } from './button.css';
+import { Box } from '../box/box';
+import { Icon } from '../icon/icon';
+import { IconNames } from '../icon/context';
+
+/**
+ * Properties for {@link Button}
+ *
+ * @public
+ */
+export interface ButtonProps {
+  size?: 'small' | 'medium';
+  variant?: 'primary' | 'secondary';
+  children: React.ReactNode;
+  disabled?: boolean;
+  iconStart?: IconNames;
+  iconEnd?: IconNames;
+}
+
+/** @public */
+export const Button = ({
+  size = 'medium',
+  variant = 'primary',
+  children,
+  disabled,
+  iconStart,
+  iconEnd,
+  ...props
+}: ButtonProps) => {
+  return (
+    <Box
+      {...props}
+      as="button"
```

Comment:
I would've expected an import from Base UI instead of this

Reply:
There's no added value from Base UI on components as simple as a button or an input. Base UI will be useful for more complex components like drawers, menus, ...

Reply:
Yes this component is by all means not finished. I highlighted the fact that it is still in progress on the home. The MUI team was debating or not to actually include a Button component. They don't have one right now. That's why I did not include it.

Reply:
Alright, we can ship as is either way. I think it's best to put high pro on bringing in some Base UI component though